pageserver/src/tenant/storage_layer/layer/tests.rs

```diff
@@ -0,0 +1,229 @@
+use futures::StreamExt;
+use tokio::task::JoinSet;
+use utils::{
+    completion::{self, Completion},
+    id::TimelineId,
+};
+
+use super::*;
+use crate::tenant::harness::TenantHarness;
+
+#[tokio::test(start_paused = true)]
+async fn residency_check_while_evict_and_wait_on_clogged_spawn_blocking() {
```

Comment:
We discussed that this test I wrote while hoping it was the cause for the problem is quite obscure because it involves racing one pending eviction with another one which re-initializes the internal state.

I would like to keep it, now that I wrote it, because this is definitely a case which can happen.

Reply:
I'll at least move it down, clean it up more.